src/cli.rs

```diff
@@ -949,6 +957,41 @@ impl Pretty {
     }
 }
 
+#[derive(Debug)]
+pub struct FormatOptions {
+    pub json_indent: usize,
+}
+
+impl Default for FormatOptions {
+    fn default() -> Self {
+        Self { json_indent: 4 }
+    }
+}
+
+impl FromStr for FormatOptions {
+    type Err = anyhow::Error;
+    fn from_str(options: &str) -> anyhow::Result<FormatOptions> {
+        let mut format_options = FormatOptions::default();
+
+        // generate a map of the specified options
+        for argument in options.split(',') {
+            let argument_parts: Vec<&str> = argument.splitn(2, ':').collect();
+            match argument_parts[..] {
+                ["json.indent", value] => {
+                    format_options.json_indent =
+                        value.parse().map_err(|_| anyhow!("Invalid indent value"))?;
```

Comment:
Since the clap will report this error, it wouldn't make a difference (I think):

```
error: Invalid value "json.indent:-5" for '--format-options <FORMAT_OPTIONS>': Invalid indent value

For more information try --help
``` 

But from perspective of code aesthetic, `.context()` looks much better than `.map_error()` 👍

Reply:
Oh yeah, looks like it'll print it the same!

The native error message is a little too technical anyway, if you pass it a negative number it says "invalid digit found in string". So maybe this is for the best.